component/loki/source/gcplog/internal/gcplogtarget/pull_target.go

```diff
@@ -137,25 +151,31 @@ func (t *PullTarget) consumeSubscription() {
 	}
 }
 
-// Labels return the model.LabelSet that the target applies to log entries.
-func (t *PullTarget) Labels() model.LabelSet {
-	lbls := make(model.LabelSet, len(t.config.Labels))
-	for k, v := range t.config.Labels {
-		lbls[model.LabelName(k)] = model.LabelValue(v)
-	}
-	return lbls
+func (t *pullTarget) Type() target.TargetType {
+	return target.GcplogTargetType
 }
 
-// Details returns some debug information about the target.
-func (t *PullTarget) Details() map[string]string {
-	return map[string]string{
-		"strategy": "pull",
-		"labels":   t.Labels().String(),
-	}
+func (t *pullTarget) Ready() bool {
+	// Return true just like all other targets.
+	// Rationale is gcplog scraping shouldn't stop because of some transient timeout errors.
+	// This transient failure can cause promtail readyness probe to fail which may prevent pod from starting.
+	// We have metrics now to track if scraping failed (`gcplog_target_last_success_scrape`).
+	return true
+}
+
+func (t *pullTarget) DiscoveredLabels() model.LabelSet {
+	return nil
+}
+
+func (t *pullTarget) Labels() model.LabelSet {
+	return t.config.Labels
+}
+
+func (t *pullTarget) Details() map[string]string {
+	return nil
```

Comment:
Same here, we've removed these Promtail methods since we don't need to implement the same interface and have no use for them at the moment.